libcst/metadata/tests/test_name_provider.py

```diff
@@ -25,7 +26,63 @@ def get_qualified_name_metadata_provider(
     return wrapper.module, wrapper.resolve(QualifiedNameProvider)
 
 
-class ScopeProviderTest(UnitTest):
+def get_qualified_names(module_str: str) -> Set[QualifiedName]:
+    _, qnames = get_qualified_name_metadata_provider(module_str)
+    return set().union(*qnames.values())
+
+
+def get_fully_qualified_names(file_path: str, module_str: str) -> Set[QualifiedName]:
+    wrapper = cst.MetadataWrapper(
+        cst.parse_module(dedent(module_str)),
+        # pyre-fixme[6]: Incompatible parameter type [6]: Expected
+        # `typing.Mapping[typing.Type[cst.metadata.base_provider.BaseMetadataProvider[
+        # object]], object]` for 2nd parameter `cache` to call
+        # `cst.metadata.wrapper.MetadataWrapper.__init__` but got
+        # `typing.Dict[typing.Type[FullyQualifiedNameProvider], object]`
+        cache={
+            FullyQualifiedNameProvider: FullyQualifiedNameProvider.gen_cache(
+                Path(""), [file_path], None
+            ).get(file_path, "")
+        },
+    )
+    return set().union(*wrapper.resolve(FullyQualifiedNameProvider).values())
```

Comment:
To make it simpler.
```suggestion
    return set(wrapper.resolve(FullyQualifiedNameProvider).values())
```

Reply:
That's what I started with but unfortunately
```
TypeError: unhashable type: 'set'
```

Reply:
```
>>> a = [[1, 2, 3], [4, 5, 6]]
>>> set(itertools.chain(*a))
{1, 2, 3, 4, 5, 6}
>>> set(y for x in a for y in x)
{1, 2, 3, 4, 5, 6}
```